addon/authenticators/oidc.js

```diff
@@ -196,6 +196,17 @@ export default BaseAuthenticator.extend({
 
     this._scheduleRefresh(expireTime, refresh_token);
 
-    return { access_token, refresh_token, userinfo, id_token, expireTime };
+    this._currentTokenData = {
+      access_token,
+      refresh_token,
+      userinfo,
+      id_token,
+      expireTime
+    };
+
+    if (!disableTrigger) {
```

Comment:
Not really happy about this one.. should this not be triggered in the places where the session actually is updated? This function does not update the session. What do you think @anehx @czosel

Reply:
Yes I agree. This should be done in `_scheduleRefresh`. However, the event should only be triggered if the refresh succeeds..

Reply:
I've put it here as it's the only place that all refresh/updates have in common. Otherwise the code would need to be duplicated. This way, the data flow is much more explicit and clear, IMHO.

It was in `_scheduleRefresh` before, but then was forgotten in `_refresh`, which lead to this bug in the first place

Reply:
Well then the real problem is, that refresh is done here without using `_scheduleRefresh`..

Reply:
I thought the same, but there we'd lose track of the retry count

Reply:
Then set it as property on the authenticator